### PR TITLE
BaseTab class improvements

### PR DESCRIPTION
**Pull request by alexey-yaremenko**
added tab's root VisualElement instead of TemplateContainer & root VisualElement is cached to make it easily accessible in derived classes

### CODE REVIEW

File: Editor/Plugins/SettingsWindow/BaseTab.cs [hunk 1]
```diff
@@ -5,11 +5,16 @@ namespace StansAssets.Foundation.Editor
...
     {
+        protected readonly VisualElement m_Root;
+
         protected BaseTab(string uxmlPath)
         {
             // Import UXML
             var visualTree = AssetDatabase.LoadAssetAtPath<VisualTreeAsset>(uxmlPath);
-            Add(visualTree.CloneTree());
+            var template = visualTree.CloneTree();
+            m_Root = template[0];
```
Comment (stan-osipov) [inline Editor/Plugins/SettingsWindow/BaseTab.cs]: `visualTree.CloneTree(this);` - Is a much better option :)

Comment (alexey-yaremenko) [reply Editor/Plugins/SettingsWindow/BaseTab.cs]: I agree, that's a much more compact solution.
but we anyway need to cache m_Root for derived classes

Comment (stan-osipov) [reply Editor/Plugins/SettingsWindow/BaseTab.cs]: I think you are right. In this case that doesn't really matter since the next step we would do the same.

Comment (alexey-yaremenko) [reply Editor/Plugins/SettingsWindow/BaseTab.cs]: false. we can do much better than this :)

**Comment by alexey-yaremenko** [comment]
@stan-osipov , committed final version. it's super compact.
please, take a look

**Pull request merged**